data-prepper-expression/src/main/java/org/opensearch/dataprepper/expression/OperatorConfiguration.java

```diff
@@ -269,4 +270,178 @@ public GenericInSetOperator inSetOperator() {
+                    Double.class, (lhs, rhs) -> (Long) lhs * (Double)rhs
+                );
+        final Map<Class<? extends Number>, BiFunction<Object, Object, Number>> doubleOperations =
+                Map.of(
+                    Integer.class, (lhs, rhs) -> (Double) lhs * (Integer) rhs,
+                    Float.class, (lhs, rhs) -> (Double) lhs * (Float) rhs,
+                    Long.class, (lhs, rhs) -> (Double) lhs * (Long) rhs,
+                    Double.class, (lhs, rhs) -> (Double) lhs * (Double) rhs
+                );
+
+        operandsToOperationMap.put(Integer.class, intOperations);
+        operandsToOperationMap.put(Float.class, floatOperations);
+        operandsToOperationMap.put(Long.class, longOperations);
+        operandsToOperationMap.put(Double.class, doubleOperations);
+
+        return new ArithmeticBinaryOperator(DataPrepperExpressionParser.MULTIPLY, operandsToOperationMap);
+    }
+
+    @Bean
+    public ArithmeticBinaryOperator divideOperator() {
+        final Map<Class<? extends Number>, Map<Class<? extends Number>, BiFunction<Object, Object, Number>>>
+                operandsToOperationMap = new HashMap<>();
+        final Map<Class<? extends Number>, BiFunction<Object, Object, Number>> intOperations =
+                Map.of(
+                        Integer.class, (lhs, rhs) -> ((double)(int)lhs) /((double)(int)rhs),
+                        Float.class, (lhs, rhs) -> (float)(int)lhs / (Float) rhs,
+                        Long.class, (lhs, rhs) -> (double)(int)lhs / (Long) rhs,
+                        Double.class, (lhs, rhs) -> (double)(int)lhs / (Double) rhs
+                );
+        final Map<Class<? extends Number>, BiFunction<Object, Object, Number>> floatOperations =
+                Map.of(
+                    Integer.class, (lhs, rhs) -> (float)lhs / (float)(int)rhs,
+                    Float.class, (lhs, rhs) ->  (float)lhs / (float)rhs,
+                    Long.class, (lhs, rhs) -> (float)lhs / (float)(long)rhs,
+                    Double.class, (lhs, rhs) -> (double)(float)lhs / (double)rhs
+                );
+        final Map<Class<? extends Number>, BiFunction<Object, Object, Number>> longOperations =
+                Map.of(
+                    Integer.class, (lhs, rhs) -> (double)(long)lhs / (double)(int)rhs,
+                    Float.class, (lhs, rhs) -> (float)(long)lhs / (Float)rhs,
+                    Long.class, (lhs, rhs) -> (double)(long)lhs / (double)(long)rhs,
+                    Double.class, (lhs, rhs) -> (double)(long)lhs / (double)rhs
+                );
+        final Map<Class<? extends Number>, BiFunction<Object, Object, Number>> doubleOperations =
+                Map.of(
+                    Integer.class, (lhs, rhs) -> (double)lhs / (double)(int)rhs,
+                    Float.class, (lhs, rhs) -> (double)lhs / (Float) rhs,
+                    Long.class, (lhs, rhs) -> (double)lhs / (double)(long)rhs,
```

Comment:
Why is the `lhs` cast to only a double where the `rhs` is cast to a `long` and then a `double`?

Reply:
Because LHS is always double (`final Map<Class<? extends Number>, BiFunction<Object, Object, Number>> doubleOperations`) and RHS has entries for 4 different types.